core/src/com/unciv/logic/civilization/NotificationActions.kt

```diff
@@ -82,13 +82,30 @@ class CityAction(private val city: Vector2 = Vector2.Zero) : NotificationAction
 }
 
 /** enter diplomacy screen */
-class DiplomacyAction(
-    private val otherCivName: String = "",
-    private var showTrade: Boolean = false
-) : NotificationAction {
+class DiplomacyAction : NotificationAction {
```

Comment:
I had to look closely, but found no flaw. Got unusual deserialization, but this won't break it. Strange kotlin quirk -> just leaving the val's in the primary constructor and adding one for `otherCiv: Civilization` won't work because the existence of a secondary constructor changes the rules for the primary and Json won't see a no-args constructor where it is implicitly made visible to Java reflection before... Therefore that much boilerplate.

Still, will people ever click twice for the same diplomacy notification in the same game session? Seemed like overkill, but why not.